R/edgeThresh.R

```diff
@@ -0,0 +1,461 @@
+    #     c(cR, .checkArgs(silent,      like = logical(1), checkSize = TRUE))
+    # cR <-
+    #     c(cR, .checkArgs(writeLog,    like = logical(1), checkSize = TRUE))
+
+    # if (length(cR) > 0) {
+    #     stop(cR)
+    # }
+    # Lmax cannot be more than graph size
+    # Validate the Lmax parameter
+    if (!missing(Lmax) &&
+        (Lmax > (numVertices <- igraph::vcount(EGG)))) {
+        stop(paste0("Lmax cannot be more than graph size."))
+    }
+    if (!missing(Lmax) && (Lmax < 2)) {
+        stop(paste0("Lmax cannot be less than 2."))
+    }
+    # Validate that the starting graph's largest connected component (weakly)
+    if (max(igraph::components(EGG, "weak")$csize) < Lmax) {
+        stop(paste0(
+            "Largest connected component in given graph has size
+            smaller than Lmax."
+        ))
+    }
+    # ((Report if graph already has delta attributes?))
+    # ((Is attribute check enough or are there other attributes?))
+    if (!is.null(igraph::graph.attributes(EGG)$delta)) {
+        stop(paste0("Graph already has attributes, remove and retry."))
+    }
+    # ((Should extremely large N and Q be tested for? Maybe a warning should be
+    # printed))
+
+    # ==== Permute and find deltas==============================================
+
+    # Use the helper function ".permuteGraph" to create permutations,
+    # which takes a copy of EGG, and returns a random
+    # graph as an igraph object
+
+    # Create a data frame to hold each permuted graph's deltas
+    masterDataFrame <-
+        data.frame(matrix(
+            vector(),
+            nrow = N,
+            ncol = (Lmax - Lmin + 1),
+            dimnames = list(N = 1:N, ords = (sapply(Lmin:Lmax, function(x)
+                paste0("ORD", x))))
+        ))
+
+    # Number of steps/increments, and their values
+    nQuant <- min(igraph::ecount(EGG), 100)
+    edgeWeights <- igraph::E(EGG)$weight
+    stepDeltas <- stats::quantile(edgeWeights, seq(0, 1, (1 / nQuant)))
+
+    # Permute and calculate deltas
+    for (i in 1:N) {
+        # Permute graph with given Q
+        cat(sprintf("\nCreating permutation %d of %d\n", i, N))
+        currentGraph <- .permuteGraph(EGG, Q)
+
+        cat(sprintf("\nCalculating deltas for graph %d of %d\n", i, N))
+
+        for (j in 1:nQuant) {
+            .pBar(j, nQuant, 80)
+            # Find edges that have weights lower than current delta and delete
+            edgesToRemove = igraph::E(currentGraph)[igraph::E(currentGraph)$weight < stepDeltas[j]]
+            currentGraph = igraph::delete.edges(currentGraph, edgesToRemove)
+
+            # Find the size of the largest strongly connected component, ord
+            ord <- max(igraph::components(currentGraph, "strong")$csize)
+
+            # If largest component has size less than Lmax, continue incrementing δ
+            # else, stop and record δ in the vector, and increase ord
+            # by 1
+            if (ord < Lmin) {
+                break
```

Comment:
I think you mean next here (unless you wanted to exit the for loop when ord < Lmin)